packages/electrumAnalyzer/electrumAnalyzer.0.3.4/opam

```diff
@@ -0,0 +1,24 @@
+opam-version: "1.2"
+maintainer: "David Chemouil <david.chemouil+electrum@onera.fr>"
+authors: ["David Chemouil" "Julien Brunel" "Denis Kuperberg"]
+homepage: "https://forge.onera.fr/projects/electrum"
+bug-reports: "https://forge.onera.fr/projects/electrum"
+license: "GPL-3+"
+dev-repo: "https://git.onera.fr/electrum.git"
+build: [
+  ["./configure" "--prefix=%{prefix}%"]
+  [make]
+]
+install: [make "install"]
+remove: [
+  ["rm" "-f" "%{bin}%/electrumAnalyzer"] {os = "darwin" | os = "linux"}
+]
+depends: [
+  "batteries" {build}
+  ("cmdliner" {build & = "0.9.7"} | "cmdliner" {build & = "0.9.6"} |
+   "cmdliner" {build & = "0.9.5"})
+  "menhir" {build & >= "20150720"}
+  "ocamlfind" {build & >= "1.5.5"}
+  "pprint" {build}
```

Comment:
On Tue, 3 Nov 2015 01:12:17 -0800, David Chemouil wrote:
> 
> I meant running oasis2opam on an updated release called 0.3.4+002. There the
> _oasis file doesn't require this.

I do not understand.  I removed the “FindlibVersion: >= 1.5.5” from your _oasis file and the lower bound on ocamlfind is not generated anymore.

> Anyway, sorry to bother you and thanks a lot for helping me here, I still have one question (I'm not used to Github or Opam packaging at all yet). There is something that I don't grasp in the expected packaging process: it looks like people put their opam configuration in their repo. But then every rebutted pull request may incur modifications to this configuration. And then, in Github terms, it seems you have to create yet another release (a new Git tag) and then update your corresponding pull request. So it seems you may have to create many tags for exactly the same piece of functionality until reaching a state where your pull request is accepted.

As Thomas said, putting an opam directory in you repo eases package pinning.  What I usually do is to use the Dynamic mode of oasis for development — so I do not have to commit an ever changing setup.ml (and other generated files) but then the opam mandates a dependency on oasis — and, when I do a release I add a tarball to it with a setup.ml fully independent of oasis (as well as other generated files) so that the final user does not need to install oasis.

>   * or submit an untagged tarball through opam-publish until it's accepted, but then once it's accepted you can at last tag it (and so submit, again, the tagged version).

That's what I do and I hope to have shown it has other advantages.

> Besides, when using oasis2opam, can we totally get rid of opam-publish prepare
> and just issue opam-publish submit when it's ready?

Yes — with the caveat that opam-publish does not like the findlib files (listing the ocamlfind libraries that the package provides) that oasis2opam generates.